### ISSUE TITLE
What is the effect of replacing multi-threaded pipelines with asynchronous queues?

### ISSUE THREAD

**Issue by jbang2004** [KoljaB/RealtimeTTS#33]
Out of pure curiosity, I want to know if asynchronous programming would be helpful in processing this project?

**Comment by KoljaB** [comment]
Yes, quite likely this project would benefit from removing most threads and processes and switching completely to asyncio, same goes for RealtimeSTT. It will be hard to implement since some calls are inherently blocking (like inferencing the synthesis with CoquiEngine). I'm not very familiar with python async programming, so I'm not even sure if it's possible. Currently I'm just to involved with other projects (new Linguflex version) to put in the work.
